### PR TITLE
amp integration

### PR DESCRIPTION
**Pull request by protonate**
## Type of change
- [x] Feature
## Description of change

An example of integrating Prebid.js into an AMP page to fill `<amp-ad>` components with Prebid demand.

### CODE REVIEW

File: integrationExamples/gpt/amp/amp_page.html [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  <head>
+      <!--
+      This is an example AMP page. Set the  amp-3p-iframe-src meta tag to your
+      x-domain host for remote.html and set your content, `amp-ad` and other components.
+      -->
+    <meta charset="utf-8">
+    <title>Hello, AMPs</title>
+    <link rel="canonical" href="http://example.ampproject.org/article-metadata.html" />
+    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+    <meta name="amp-3p-iframe-src" content="https://amp.prebidapp.com:5000/remote.html">
```
Comment (mkendall07) [inline integrationExamples/gpt/amp/amp_page.html]: let's make the publisher hosted content obvious with a macro or something.

File: integrationExamples/gpt/amp/remote.html [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+     */
+
+    (function() {
+var v = location.search.substr(1);
+if (!(/^\d+(-canary)?$/.test(v))) return;
+var u = 'https://3p.ampproject.net/'+encodeURIComponent(v)+'/f.js';
+document.write('<script'+' src="'+encodeURI(u)+'"><'+'/script>');
+})();
+</script>
+<script>
```
Comment (mkendall07) [inline integrationExamples/gpt/amp/remote.html]: Let's add a `<!-- start Prebid.js AMP integration -->` comment

File: integrationExamples/gpt/amp/remote.html [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+    var date = new Date().getTime();
+
+    // wrap the rest of the setup in a function that will be called by the
+    // AMP `draw3p` hook
+    function loadPrebidJS() {
+
+        setTimeout(function() {
+            return initAdserver;
+        }(), PREBID_TIMEOUT);
```
Comment (mkendall07) [inline integrationExamples/gpt/amp/remote.html]: Do we need this timeout ?

File: integrationExamples/gpt/amp/remote.html [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+
+        pbjs.que.push(function () {
+
```
Comment (mkendall07) [inline integrationExamples/gpt/amp/remote.html]: nit: extra line breaks / spacing.

File: integrationExamples/gpt/amp/remote.html [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+                },
+                timeout: 3000
+            });
+        });
+
+        // load Prebid.js
+       (function () {
+            var d = document, pbs = d.createElement("script"), pro = d.location.protocal;
+            pbs.type = "text/javascript";
+            pbs.src = 'https://localhost:9999/build/dev/prebid.js';
```
Comment (mkendall07) [inline integrationExamples/gpt/amp/remote.html]: can we load this in the head?

File: integrationExamples/gpt/amp/remote.html [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+  <script>
+    draw3p(function(config, done) {
+      if (typeof window.context.master.adCalls === 'undefined') {
+        window.context.master.adCalls = [];
+      }
+        if (window.context && window.context.isMaster) {
+            loadPrebidJS();
+        }
+        window.context.master.adCalls.push(setTargeting.bind(null, config, done));
+    }, ['doubleclick'], ['prebidapp.com']);
```
Comment (mkendall07) [inline integrationExamples/gpt/amp/remote.html]: What's the meaning `prebidapp.com` here?

File: integrationExamples/gpt/amp/remote.html [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+        setTimeout(function() {
+            return initAdserver;
+        }(), PREBID_TIMEOUT);
+
+
+        pbjs.que.push(function () {
+
+            pbjs.logging = true;
+
+            var adUnits = [
```
Comment (mkendall07) [inline integrationExamples/gpt/amp/remote.html]: can we clean this up a little bit so all publisher defined variables are declared in one spot and that's the only code that has to change on the page?

**Comment by protonate** [comment]
@mkendall07 notes addressed

**Pull request merged**